javarosa/src/main/java/org/javarosa/core/io/StreamsUtil.java

```diff
@@ -110,4 +78,71 @@ public OutputIOException(IOException internal) {
             super(internal);
         }
     }
+
+    public static byte[] inputStreamToByteArray(InputStream input) throws IOException {
+        byte[] buffer = new byte[8192];
+        int bytesRead;
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+        while ((bytesRead = input.read(buffer)) != -1) {
+            output.write(buffer, 0, bytesRead);
+        }
+        return output.toByteArray();
+    }
+
+    /**
+     * Writes input stream to output stream in a buffered fasion, but doesn't
+     * close either stream.
+     */
+    public static void writeFromInputToOutputUnmanaged(InputStream is,
+                                                       OutputStream os) throws IOException {
+        byte[] buffer = new byte[8192];
+        int count = is.read(buffer);
+        while (count != -1) {
+            os.write(buffer, 0, count);
+            count = is.read(buffer);
+        }
+    }
+
+    /**
+     * Write is to os and close both
+     */
+    public static void writeFromInputToOutputNew(InputStream is, OutputStream os) throws IOException {
+        writeFromInputToOutputNew(is, os, null);
+    }
+
+    /**
+     * Write is to os and close both
+     */
+    public static void writeFromInputToOutputNew(InputStream is, OutputStream os, StreamReadObserver observer) throws IOException {
+        byte[] buffer = new byte[8192];
```

Comment:
I think it is what the default buffer size is for similar library implementations of code like this.